File: nfsw/epilogue.py
# -*- coding: utf-8 -*-
#
#   SPDX-License-Identifier: ISC
#
#   Copyright (C) 2019 rsiddharth <s@ricketyspace.net>
#
#   This file is part of dingy.
#

from nfsw.auth import login_required
from nfsw.redis import redis


from flask import (
    Blueprint, render_template, redirect, url_for
)

bp = Blueprint('epilogue', __name__)


@bp.route('/epilogue', endpoint='epilogue')
@login_required
def epilogue():
    r = redis()

    if r.exists('epilogue:done'):
        return redirect(url_for('io'))

    r.set('epilogue:done', 1)

    return render_template('epilogue.html')

File: nfsw/static/epilogue.js
/**
 * SPDX-License-Identifier: ISC
 *
 * Copyright (C) 2019 rsiddharth <s@ricketyspace.net>
 *
 * This file is part of nfsw.
 */

document.addEventListener('DOMContentLoaded', function() {
    var ep_wrapper = document.getElementsByClassName('epilogue-wrapper')[0];
    ep_wrapper.style = 'display:block';

    window.setTimeout(
        function() {
            window.location.href = "/io";
        }, 5000
    );
});
